File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> unique = new HashSet<>();
        for (int num:nums){
            unique.add(num);
        }
        int res = 0;
        for (int i =0;i<nums.length;i++){
            int check = nums[i];
            if (!unique.contains(check-1)){
                int length =1;
                while(unique.contains(check+length)){
                    length ++;
                }
                res=Math.max(length,res);
            }
        }
        return res;
        
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-6.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prev = [0] * (len(nums))
        after = [0] * (len(nums))
        res = [0] * (len(nums))

        prev[0] = after[len(nums)-1] = 1
        for i in range(1,len(nums)):
            prev[i] = nums[i-1] * prev[i-1]
        print(prev)
        for i in range(len(nums)-2,-1,-1):
            after[i] = nums[i+1] * after[i+1]
        print(after)
        for i in range(len(nums)):
            res[i] = prev[i] * after[i]
        return res
        